src/components/specific/LocationSearchModal/LocationSearchModal.js

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useRef, useState } from "react";
+import { Modal } from "../../common/Modal";
+import { Button } from "../../common/Button";
+import "./LocationSearchModal.css";
+
+export function LocationSearchModal({ closeModal, addData }) {
+    const [ enteredLocation, setEnteredLocation ] = useState(null);
+    const [ possibleLocations, setPossibleLocations ] = useState([]);
+    const [ selectedLocations, setSelectedLocations ] = useState([]);
+    const locationInputRef = useRef();
+
+    useEffect(() => {
+        if(enteredLocation !== null) {
+            fetch(`https://nominatim.openstreetmap.org/search?q=${encodeURIComponent(enteredLocation)}&format=jsonv2&addressdetails=1`)
+            .then(response => {
+                return response.json();
+            })
+            .then(data => {
+                // We will filter out all duplicate locations (ones with same name and displayName)
+                const data_hash = {};
+                data = data.filter(item => {
+                    if(data_hash[`${item?.name}${item?.displayName}`]) return false;
+                    else {
+                        data_hash[`${item?.name}${item?.displayName}`] = true;
+                        return true;
+                    }
+                })
+
+                // We will take only the first 4 matches
+                data = data.slice(0, 4);
```

Comment:
add null pointer

Reply:
Fixed in commit "Update dot operator to null operator to avoid potential blockers".